### DISCUSSION TITLE
Lab2 fail to co-sim

### DISCUSSION THREAD

**Discussion by junshiuanhsieh** [bol-edu/HLS-SOC-Discussions#73] [Q&A] [answered]
我在做Lab2 AXI-master時，csim/csyn都有過之後，co-sim時就不會過。我看lab2 github上的tutorial上面沒有提到要co-sim，但老師上課時又有特別講解co-sim，想請問一下需要co-sim嗎，如果需要，出現錯誤可能是什麼原因？
btw Stream就不會有這個問題

以下附上co-sim出現的error

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/58873985/72d0cee1-6afc-4cbd-9908-f463b3cbdd51)

**Comment by holyuming** [comment]
同問，這次workbook裡面並沒有說到需要co-sim，但是繳交檔案需要cosim.rpt?

**Comment by gene5487** [comment]
我有遇過相同的co-sim error，依照workbook的1.2.2部分，設定Directive即可解決。

Reference:
[https://github.com/bol-edu/HLS-SOC-Discussions/discussions/53](url)

**Reply by emma16163** [reply, marked as answer]
可以參考我的Directive設定
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/75108089/cd58e63e-cf18-4032-adbf-dfd3bbee8ddb)

**Reply by junshiuanhsieh** [reply]
感謝大家 問題解決了
github tutorial pdf上面的圖片寫depth=128，檔案裡面的tcl也是寫128，但後來發現depth得是600才行

**Comment by holyuming** [comment]
沒錯，我一開始設Depth = 1000後面也會有errors，600才沒問題
